### ISSUE TITLE
S3 failure when item not yet created

### ISSUE THREAD

**Issue by siznax** [internetarchive/draintasker#6]
@kngenie: Draintasker is trying to upload the second file to the item just created, and item wasn't created yet.

```
ERROR: S3 PUT failed with response_code: 404 at 2020-03-26T22:48:22PDT
BLOCK: sleep for 120 seconds...
```

@kngenie: thinking now, draintasker could just keep uploading files with "create item" option, rather than being cautious

**Comment by siznax** [comment]
Need to give this some more thought and do some testing.

**Comment by siznax** [comment]
1: draintasker does not upload WARC files immediately, but first upload a tiny text file to a new item and wait for the item to become available (== GET succeeds with 200), before it starts uploading WARC files. My understanding is draintasker does not need to wait -- it can just upload WARC files with `x-archive-auto-make-bucket=1`

2: I believe that you can do that, and I also believe that will help speed things up.

3: ah, i see. yes, that seems like an excellent suggestion.

**Comment by siznax** [comment]
This should be fixed with 447c338

**Comment by siznax** [comment]
Not finished yet. We also need disable checking if bucket exists if `auto-make-bucket` succeeds.

```
# initial upload of MANIFTEST
  --upload-file /1/crawling/youtube/sink/YTV-20200601125736-crawl819/MANIFEST
   http://s3.us.archive.org/YTV-20200601125736-crawl819/MANIFEST.txt
  -o /var/tmp/curl.out.13120
curl finished with status: 0
SUCCESS: S3 PUT succeeded with response_code: 200
item/bucket created successfully: YTV-20200601125736-crawl819

Checking if bucket YTV-20200601125736-crawl819 exists
curl -vv --include --location
  --header authorization: LOW kikcjvxLFuXjeXb1:Xeww747yjP8Gyn8S
  --write-out %{http_code} %{size_upload} %{time_total} --head
   http://s3.us.archive.org/YTV-20200601125736-crawl819/
  -o /var/tmp/curl.out.13120
curl finished with status: 0
ERROR: S3 PUT failed with response_code: 404 at 2020-06-01T16:50:45PDT
BLOCK: sleep for 30 seconds...
done sleeping at 2020-06-01T16:51:15PDT

Checking if bucket YTV-20200601125736-crawl819 exists
RETRY attempt (1) Mon Jun 1 16:51:15 PDT 2020
curl -vv --include --location
  --header authorization: LOW kikcjvxLFuXjeXb1:Xeww747yjP8Gyn8S
  --write-out %{http_code} %{size_upload} %{time_total} --head
   http://s3.us.archive.org/YTV-20200601125736-crawl819/
  -o /var/tmp/curl.out.13120
curl finished with status: 0
ERROR: S3 PUT failed with response_code: 404 at 2020-06-01T16:51:19PDT
BLOCK: sleep for 30 seconds...
done sleeping at 2020-06-01T16:51:49PDT

...
```

We want to prevent "Checking if bucket exists"
